source/plugins/ruby/constants.rb

```diff
@@ -144,4 +144,16 @@ class Constants
   POD_MDM_EMIT_STREAM_BATCH_SIZE = 5000 # each record is 200 bytes, 5k records ~2MB
   # only used in windows in AAD MSI auth mode
   IMDS_TOKEN_PATH_FOR_WINDOWS = "c:/etc/imds-access-token/token"
+
+  # extension settings and these should match with DCR extension settings
+  EXTENSION_SETTINGS = "extensionSettings"
+  EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS = "dataCollectionSettings"
+  EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_INTERVAL = "interval"
+  EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_NAMESPACE_FILTERING_MODE = "nameSpaceFilteringMode"
+  EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_NAMESPACES_FILTERING_MODES = ["off", "include", "exlcude"]
+  EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_NAMESPACES = "nameSpaces"
```

Comment:
naming appears inconsistent. is "dataCollection" appended only for settings? why not dataCollectionInterval, dataCollectionNameSpaceFilteringMode, etc?

Reply:
> naming appears inconsistent. is "dataCollection" appended only for settings? why not dataCollectionInterval, dataCollectionNameSpaceFilteringMode, etc?
> dataCollectionSettings is top level object name and remaining fields inside that, so adding prefix for all the fields is not required as we discussed.